src/Kafka/KafkaMessageBrowser/KafkaMessageBrowser.tsx

```diff
@@ -0,0 +1,365 @@
+        send({ type: "setTimestamp", value })
+      }
+      setEpoch={(value: number | undefined) =>
+        send({ type: "setEpoch", value })
+      }
+      setLatest={() => send({ type: "setLatest" })}
+      setLimit={(value: number) => send({ type: "setLimit", value })}
+      refresh={() => send({ type: "refresh" })}
+      selectMessage={(message: Message) =>
+        send({ type: "selectMessage", message })
+      }
+      deselectMessage={() => send({ type: "deselectMessage" })}
+    />
+  );
+};
+
+export type KafkaMessageBrowserConnectedProps = {
+  isFirstLoad: boolean;
+  isNoData: boolean;
+  isRefreshing: boolean;
+  requiresSearch: boolean;
+  selectedMessage: Message | undefined;
+  lastUpdated: Date | undefined;
+  response: MessageApiResponse | undefined;
+  partition: number | undefined;
+  limit: number;
+  filterOffset: number | undefined;
+  filterTimestamp: number | undefined;
+  setPartition: (value: number | undefined) => void;
+  setOffset: (value: number | undefined) => void;
+  setTimestamp: (value: Date | undefined) => void;
+  setEpoch: (value: number | undefined) => void;
+  setLatest: () => void;
+  setLimit: (value: number) => void;
+  refresh: () => void;
+  selectMessage: (message: Message) => void;
+  deselectMessage: () => void;
+};
+export const KafkaMessageBrowserConnected: VoidFunctionComponent<
+  KafkaMessageBrowserConnectedProps
+> = ({
+  isFirstLoad,
+  isNoData,
+  isRefreshing,
+  requiresSearch,
+  selectedMessage,
+  response,
+  partition,
+  limit,
+  filterOffset,
+  filterTimestamp,
+  setPartition,
+  setOffset,
+  setTimestamp,
+  setEpoch,
+  setLatest,
+  setLimit,
+  refresh,
+  selectMessage,
+  deselectMessage,
+}) => {
+  const { t } = useTranslation("message-browser");
+  const [defaultTab, setDefaultTab] =
+    useState<MessageDetailsProps["defaultTab"]>("value");
+
+  const columnLabels: { [key in typeof columns[number]]: string } = useMemo(
+    () =>
+      ({
+        partition: t("field.partition"),
+        offset: t("field.offset"),
+        timestamp: t("field.timestamp"),
+        key: t("field.key"),
+        value: t("field.value"),
+        headers: t("field.headers"),
+      } as const),
+    [t]
+  );
+
+  switch (true) {
+    case isFirstLoad:
+      return <Loading />;
+    case isNoData:
+      return <NoDataEmptyState onRefresh={refresh} />;
+    default:
+      return (
+        <Drawer isInline={true} isExpanded={selectedMessage !== undefined}>
+          <DrawerContent
+            panelContent={
+              <MessageDetails
+                message={selectedMessage}
+                defaultTab={defaultTab}
+                onClose={deselectMessage}
+              />
+            }
+          >
+            <Toolbar
+              className={"mas-KafkaMessageBrowser-Toolbar"}
+              data-testid={"message-browser-toolbar"}
+            >
+              <ToolbarContent>
+                <ToolbarToggleGroup toggleIcon={<FilterIcon />} breakpoint="xl">
```

Comment:
Since this breakpoint is dependent on the viewport size, and since the drawer provides the ability to alter the available area for the toolbar without triggering this responsive behavior, would it work to have this value conditionally set to `2xl` when the drawer is visible to force the toggle to display sooner?

Reply:
👍